src/anomalib/data/kolektor.py

```diff
@@ -0,0 +1,315 @@
+        test_split_ratio (float): Fraction of images from the train set that will be reserved for testing.
+            Defaults to 0.2
+        val_split_mode (ValSplitMode): Setting that determines how the validation subset is obtained.
+        val_split_ratio (float): Fraction of train or test images that will be reserved for validation.
+            Defaults to 0.5
+        seed (int | None, optional): Seed which may be set to a fixed value for reproducibility.
+    """
+
+    def __init__(
+        self,
+        root: Path | str,
+        image_size: int | tuple[int, int] | None = None,
+        center_crop: int | tuple[int, int] | None = None,
+        normalization: str | InputNormalizationMethod = InputNormalizationMethod.IMAGENET,
+        train_batch_size: int = 32,
+        eval_batch_size: int = 32,
+        num_workers: int = 8,
+        task: TaskType = TaskType.SEGMENTATION,
+        transform_config_train: str | A.Compose | None = None,
+        transform_config_eval: str | A.Compose | None = None,
+        test_split_mode: TestSplitMode = TestSplitMode.FROM_DIR,
+        test_split_ratio: float = 0.2,
+        val_split_mode: ValSplitMode = ValSplitMode.SAME_AS_TEST,
+        val_split_ratio: float = 0.5,
+        seed: int | None = None,
+    ) -> None:
+        super().__init__(
+            train_batch_size=train_batch_size,
+            eval_batch_size=eval_batch_size,
+            num_workers=num_workers,
+            test_split_mode=test_split_mode,
+            test_split_ratio=test_split_ratio,
+            val_split_mode=val_split_mode,
+            val_split_ratio=val_split_ratio,
+            seed=seed,
+        )
+
+        self.root = Path(root)
+
+        transform_train = get_transforms(
+            config=transform_config_train,
+            image_size=image_size,
+            center_crop=center_crop,
+            normalization=InputNormalizationMethod(normalization),
+        )
+        transform_eval = get_transforms(
+            config=transform_config_eval,
+            image_size=image_size,
+            center_crop=center_crop,
+            normalization=InputNormalizationMethod(normalization),
+        )
+
+        self.train_data = KolektorDataset(
+            task=task,
+            transform=transform_train,
+            split=Split.TRAIN,
+            root=root,
+        )
+        self.test_data = KolektorDataset(
+            task=task,
+            transform=transform_eval,
+            split=Split.TEST,
+            root=root,
+        )
+
+    def prepare_data(self) -> None:
+        """Download the dataset if not available."""
+        if (self.root).is_dir():
+            logger.info("Found the dataset.")
+        else:
+            download_and_extract_kolektor(
+                "https://go.vicos.si/kolektorsdd",
+                self.root,
+                "2b094030343c1cd59df02203ac6c57a0",
+            )
```

Comment:
If you define `DOWNLOAD_INFO` variable as given below

```python
DOWNLOAD_INFO = DownloadInfo(
    name="kolektor",
    url="https://go.vicos.si/kolektorsdd",
    hash="2b094030343c1cd59df02203ac6c57a0",
)
```

Then it would be possible to use `anomalib.data.download.download_and_extract` function as:

```python
download_and_extract(self.root, DOWNLOAD_INFO)
```

This would be inline with the rest of the datamodule implementations. For reference, here are the some of the implementations:
[mvtec-ad](https://github.com/openvinotoolkit/anomalib/blob/main/src/anomalib/data/mvtec.py#L52), [avenue](https://github.com/openvinotoolkit/anomalib/blob/main/src/anomalib/data/avenue.py#L44), [btech](https://github.com/openvinotoolkit/anomalib/blob/main/src/anomalib/data/btech.py#L38), [mvtec-3d](https://github.com/openvinotoolkit/anomalib/blob/main/src/anomalib/data/mvtec_3d.py#L49)...

Reply:
This links with the previous issue which makes the `download_and_extract` function not usable here.